### PR TITLE
GAP-1784: Enter new admin email change ownership page

### PR DESCRIPTION
**Pull request by dominicwest**
## Description

[GAP-1784](https://technologyprogramme.atlassian.net/jira/software/projects/GAP/boards/216?selectedIssue=GAP-1784)

Depends on: 
- https://github.com/cabinetoffice/gap-find-admin-backend/pull/60
- https://github.com/cabinetoffice/gap-user-service/pull/147

Summary of changes:
- Implementing the change ownership new admin email page
   - Validating email format & that an admin owns it via an endpoint
- Fixing FlexibleQuestionPageLayout width
- Implementing a confirm change ownership page
   - Calls the patch that updates ownership

## Type of change

Please check the relevant options.

- [x] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [x] Unit Test

- [ ] Integration Test (if applicable)

- [ ] End to End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:

https://github.com/cabinetoffice/gap-find-apply-web/assets/101722961/2d307bb5-724a-48db-8c44-7651ceb44dce

Screenshot 2023-11-09 at 10 55 31
Screenshot 2023-11-09 at 10 55 16
Screenshot 2023-11-09 at 10 54 58
Screenshot 2023-11-09 at 10 54 44
Screenshot 2023-11-09 at 12 06 59

# Checklist:

- [x] If I have listed depedenencies above, I have ensured that they are present in the target branch.
- [x] I have performed a self-review of my code.
- [x] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

[GAP-1784]: https://technologyprogramme.atlassian.net/browse/GAP-1784?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/admin/src/pages/super-admin-dashboard/user/[id]/index.page.tsx [hunk 1]
```diff
@@ -129,7 +129,14 @@ const UserPage = (pageData: InferProps<typeof getServerSideProps>) => {
...
                         <Link
-                          href={`/super-admin-dashboard/user/${pageData.gapUserId}/change-owner`}
+                          href={`/super-admin-dashboard/user/${
+                            pageData.gapUserId
+                          }/schemes/${
+                            scheme.schemeId
+                          }/change-owner?oldEmailAddress=${encodeURIComponent(
+                            pageData.emailAddress
+                          )}&schemeName=${encodeURIComponent(scheme.name)}
+                            `}
```
Comment (ConorFayleTCO) [inline packages/admin/src/pages/super-admin-dashboard/user/[id]/index.page.tsx:139]: can we extract this somehow? it hurts my brain

File: packages/admin/src/pages/super-admin-dashboard/user/[id]/schemes/[schemeId]/change-owner.page.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    fetchPageData,
+    handleRequest,
+    jwt: getSessionIdFromCookies(context.req),
+    onErrorMessage: 'Failed to change scheme owner.',
+    onSuccessRedirectHref: (emailAddress) =>
+      `/super-admin-dashboard/user/${userId}/schemes/${schemeId}/confirm-change-owner?newEmailAddress=${encodeURIComponent(
+        emailAddress
+      )}&oldEmailAddress=${encodeURIComponent(
+        oldEmailAddress
+      )}&schemeName=${encodeURIComponent(schemeName)}`,
```
Comment (ConorFayleTCO) [inline packages/admin/src/pages/super-admin-dashboard/user/[id]/schemes/[schemeId]/change-owner.page.tsx:59]: yeah can we write a helper method to build this url? its uggo

Comment (dominicwest) [reply packages/admin/src/pages/super-admin-dashboard/user/[id]/schemes/[schemeId]/change-owner.page.tsx:59]: yee good shout

File: packages/admin/src/pages/super-admin-dashboard/user/[id]/index.page.tsx [hunk 3]
```diff
@@ -125,16 +125,22 @@ const UserPage = (pageData: InferProps<typeof getServerSideProps>) => {
...
-                          <a className="govuk-link">Change owner</a>
-                        </Link>
-                      ),
-                    }))}
+                    rows={pageData.schemes.map((scheme) => {
+                      const queryParams = new URLSearchParams({
+                        oldEmailAddress: pageData.emailAddress,
+                        schemeName: scheme.name,
+                      }).toString();
+                      const href = `/super-admin-dashboard/user/${pageData.gapUserId}/schemes/${scheme.schemeId}/change-owner?${queryParams}`;
```
Comment (ConorFayleTCO) [inline packages/admin/src/pages/super-admin-dashboard/user/[id]/index.page.tsx:133]: nice way better

**Review by ConorFayleTCO — APPROVED** [review]

**Pull request merged**